src/pytest_mysql/factories/client.py

```diff
@@ -131,8 +131,27 @@ def mysql_fixture(request: FixtureRequest) -> MySQLdb.Connection:
         yield mysql_conn
 
         # clean up after test that forgot to fetch selected data
-        mysql_conn.store_result()
-        mysql_conn.query("DROP DATABASE IF EXISTS %s" % mysql_db)
-        mysql_conn.close()
+        try:
+            if mysql_conn.open:
+                mysql_conn.store_result()
+        except Exception as e:
```

Comment:
Catch specific Exceptions and handle them. Let the unknown exception escalate. Here we could probably catch exception that will tell us that the connection is closed, and the next "if" won't be needed.

Reply:
Right, but I'm figuring out that if not for mysql_conn is open, why is important to escalate an unknown exception?
The thing is --> if an exception here is thrown, database is not dropped